MinitexPDFProtocols/MinitexPDFProtocols.swift

```diff
@@ -9,67 +9,92 @@
 import Foundation
 import UIKit
 
-// the items that are marked required are needed to
```

Comment:
This will be a theme in a couple places, and I will provide a full example of how this will "flow down" to PDF-iOS at the end of the comments. But basically: the protocols here are expected to represent _any_ pdf renderer, and so the decisions below will be made with that in mind.
If we define things here that only PSPDFKit can understand, we're stuck for any other renderer. However, if we start with almost nothing, things that only PSPDFKit can understand can be added from _inside_ PDF-iOS with a new class that conforms to this protocol.